src/examples/using-with-cmake/host-config.cmake.in

```diff
@@ -29,5 +29,20 @@ set(MPI_Fortran_COMPILER   "@MPI_Fortran_COMPILER@"   CACHE PATH "")
 set(MPIEXEC_EXECUTABLE     "@MPIEXEC_EXECUTABLE@"     CACHE PATH "")
 set(MPIEXEC_NUMPROC_FLAG   "@MPIEXEC_NUMPROC_FLAG@"   CACHE STRING "")
 
+# CUDA
+set(ENABLE_CUDA                @ENABLE_CUDA@                  CACHE BOOL "")
+set(CUDA_TOOLKIT_ROOT_DIR      "@CUDA_TOOLKIT_ROOT_DIR@"      CACHE PATH "")
+set(CMAKE_CUDA_COMPILER        "@CMAKE_CUDA_COMPILER@"        CACHE PATH "")
+set(CMAKE_CUDA_HOST_COMPILER   "@CMAKE_CUDA_HOST_COMPILER@"   CACHE PATH "")
+set(CMAKE_CUDA_ARCHITECTURES   "@CMAKE_CUDA_ARCHITECTURES@"   CACHE STRING "")
+set(CMAKE_CUDA_FLAGS           "@CMAKE_CUDA_FLAGS@"           CACHE STRING "")
```

Comment:
Should this be wrapped in 
```cmake
if(ENABLE_CUDA) 
   ... 
endif()
```
 ?

Reply:
There is an `if()` check in the `CMakeLists.txt` to check for device. 

I was considering this, but I do not know if it is best practice to put `if()` checks in our host-configs (technically we do have an `if()` check that is spack generated in our current host-configs, but the check only matters for when spack is using the host-config).